form/_AutoCompleterMixin.js

```diff
@@ -486,6 +486,16 @@ define([
 			}
 			this.inherited(arguments);
 			aspect.after(this, "onSearch", lang.hitch(this, "_openResultList"), true);
+
+			// if the "clear" button is pressed in IE (provided by <input>), then clear the value
+			// must be async since the clearing of the input element happens after the mouseup event 
+			this.on("mouseup", function (e) {
+				setTimeout(lang.hitch(this, function() {
+					if (!e.target.value) {
+						this.set("item", null);
+					}
+				}), 0); 
+			});
```

Comment:
Can't you detect this via the oninput event instead?   That's supposed to be a catch-all event to include corner cases like this.